internal/pkg/coordinator/monitor.go

```diff
@@ -74,8 +74,10 @@ type monitorT struct {
 	leadersIndex  string
 	agentsIndex   string
 
-	policies          map[string]policyT
-	policiesCanceller map[string]context.CancelFunc
+	policies map[string]policyT
+
+	muPoliciesCanceller sync.Mutex
+	policiesCanceller   map[string]context.CancelFunc
```

Comment:
I know that an integration test checks the size of this attribute, hopefully it's not an issue

Reply:
oh, let me take a look, that seems like a bad pattern in that case.